planner/core/rule_join_reorder.go

```diff
@@ -89,6 +126,18 @@ func (s *joinReOrderSolver) optimizeRecursive(ctx sessionctx.Context, p LogicalP
 		if err != nil {
 			return nil, err
 		}
+		sort.Slice(leftJoinTabs, func(i, j int) bool {
+			if len(leftJoinTabs[j].eqCond) > 0 && leftJoinTabs[i].p.Schema().ColumnIndex(leftJoinTabs[j].eqCond[0].GetArgs()[0].(*expression.Column)) >= 0 {
```

Comment:
Please add comments for these 2 heuristic rules.

Reply:
added